sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeContentOperation.cs

```diff
@@ -128,5 +151,36 @@ private static FormPageCollection ConvertValue(IReadOnlyList<PageResult_internal
 
             return new FormPageCollection(pages);
         }
+
+        private async ValueTask<RequestFailedException> CreateExceptionForFailedOperationAsync(bool async, IReadOnlyList<FormRecognizerError> errors)
+        {
+            string errorMessage = default;
+            string errorCode = default;
+
+            if (errors.Count > 0)
+            {
+                var firstError = errors[0];
+
+                errorMessage = firstError.Message;
+                errorCode = firstError.ErrorCode;
+            }
+            else
+            {
+                errorMessage = "RecognizeContent operation failed.";
+            }
+
+            var errorInfo = new Dictionary<string, string>();
+            int index = 0;
+
+            foreach (var error in errors)
+            {
+                errorInfo.Add($"error-{index}", $"{error.ErrorCode}: {error.Message}");
```

Comment:
Maybe I am missing something here, but unsure how the map would help here? The user would still have to iterate through it like it were a list. Or maybe I am

Reply:
The map is something we pass to `CreateRequestFailedException` so it can list the errors in our exception message. It's not an object we pass to the user directly.